### DISCUSSION TITLE
OpenNHP是如何保护acdemo.opennhp.org

### DISCUSSION THREAD

**Discussion by i6asic** [OpenNHP/opennhp#1210] [Q&A] [answered]
您好

在尝试了OpenNHP保护了https://acdemo.opennhp.org 地址的Demo效果之后,我感觉非常惊奇,感叹OpenNHP的强大能力.

在我通读了官方文档之后,发现没有资料去描述和解释这个Demo是如何部署和运作的,这令我感到困惑

1. 官方文档中对于OpenNHP的部署说明,用户端需要部署NHP Agent,这与Demo并不相同
2. Demo是否像文档中描述的那样,与DNS进行了集成?

另外,除了论文link之外 是否有其他文档帮助落地和理解NHP协议?

希望可以解答疑惑,感谢感谢

**Comment by craftleon** [comment, marked as answer]
1. 官方文档中对于OpenNHP的部署说明,用户端需要部署NHP Agent,这与Demo并不相同
答：NHP server可以支持UDP与HTTPS的敲门验证。NHP Agent使用UDP发送NHP敲门请求，用于C/S架构，安全性更高。在Demo中，出于简化用户侧安装与配置的原因，没有使用NHP Agent，而使用了B/S架构。
2. Demo是否像文档中描述的那样,与DNS进行了集成?
答：在Demo中，我们是对一个网页访问进行隐身保护，本身与DNS无关。与DNS集成是NHP的一个使用场景，用来对特定用户开放DNS解析服务而对陌生用户隐身，以保护DNS服务器。
3. 除了论文link之外 是否有其他文档帮助落地和理解NHP协议?
答：文档我们会持续更新，请保持关注。

**Reply by Norton-Lin** [reply]
我很好奇一些问题，在这个Demo的实现中，我推测应该是没有插件的的存在吧。那么这里的NHP敲门请求等客户端应该执行的工作，是如何实现的呢？用了一些js脚本之类？

**Reply by craftleon** [reply]
Demo的逻辑比较简单，用户在访问登录页https://demologin.opennhp.org/ 时会拿到js脚本，用来向NHP server发送敲门请求以访问目的网页https://acdemo.opennhp.org/ 。NHP server会调用example插件对用户敲门请求进行验证（见代码）并放行。

**Comment by jerry-0824** [comment]
你好，关于您提到的问题，稍后我们会在 https://opennhp.org/ 对文档进行更新。
感谢您的关注～
